src/sync/deployment.ts

```diff
@@ -49,12 +72,48 @@ const getTransformedDeployments = async (deployments, gitHubInstallationClient:
 	});
 
 	const transformedDeployments = await Promise.all(transformTasks);
-	return transformedDeployments
-		.filter(deployment => !!deployment)
+	return (transformedDeployments
+		.filter(deployment => !!deployment) as JiraDeploymentBulkSubmitData[])
 		.map(deployment => deployment.deployments)
 		.flat();
 };
 
+const saveDeploymentsForLaterUse = async (deployments: FetchDeploymentResponse["deployments"], gitHubBaseUrl: string, logger: Logger) => {
+	try {
+
+		const successDeployments: FetchDeploymentResponse["deployments"] = deployments.filter(d => (d.statuses?.nodes.some(n => n.state === "SUCCESS")));
```

Comment:
/nit Your can drop the type for `const successDeployments: FetchDeploymentResponse["deployments"]` since its inferred from the filter response.

Reply:
Good call, it must be my glitch on my vim ts settigs that doesn't show up correctly, so I added the extra types.
Sorry since this is already deployed, I am leaving this as it is right now. I may remember to remove it in next PR.